error: invalid path URI: expected token '/' but found 'a' at index 0
 --> $DIR/route-path-bad-syntax.rs:7:8
  |
7 | #[get("a")] //~ ERROR invalid path URI
  |        ^^
  |
  = help: expected path in origin form: "/path/<param>"

error: invalid path URI: expected token '/' but none was found at index 0
  --> $DIR/route-path-bad-syntax.rs:11:8
   |
11 | #[get("")] //~ ERROR invalid path URI
   |        ^
   |
   = help: expected path in origin form: "/path/<param>"

error: invalid path URI: expected token '/' but found 'a' at index 0
  --> $DIR/route-path-bad-syntax.rs:15:8
   |
15 | #[get("a/b/c")] //~ ERROR invalid path URI
   |        ^^^^^^
   |
   = help: expected path in origin form: "/path/<param>"

error: paths cannot contain empty segments
  --> $DIR/route-path-bad-syntax.rs:19:7
   |
19 | #[get("/a///b")] //~ ERROR empty segments
   |       ^^^^^^^^
   |
   = note: expected '/a/b', found '/a///b'

error: query cannot contain empty segments
  --> $DIR/route-path-bad-syntax.rs:23:10
   |
23 | #[get("/?bat&&")] //~ ERROR empty segments
   |          ^^^^^

error: query cannot contain empty segments
  --> $DIR/route-path-bad-syntax.rs:26:10
   |
26 | #[get("/?bat&&")] //~ ERROR empty segments
   |          ^^^^^

error: paths cannot contain empty segments
  --> $DIR/route-path-bad-syntax.rs:29:7
   |
29 | #[get("/a/b//")] //~ ERROR empty segments
   |       ^^^^^^^^
   |
   = note: expected '/a/b', found '/a/b//'

error: component contains invalid URI characters
  --> $DIR/route-path-bad-syntax.rs:35:10
   |
35 | #[get("/?????")] //~ ERROR invalid URI characters
   |          ^^^^
   |
   = note: components cannot contain '%' and '+' characters

error: invalid path URI: expected EOF but found '#' at index 3
  --> $DIR/route-path-bad-syntax.rs:39:11
   |
39 | #[get("/!@#$%^&*()")] //~ ERROR invalid path URI
   |           ^^^^^^^^^
   |
   = help: expected path in origin form: "/path/<param>"

error: component contains invalid URI characters
  --> $DIR/route-path-bad-syntax.rs:43:9
   |
43 | #[get("/a%20b")] //~ ERROR invalid URI characters
   |         ^^^^^
   |
   = note: components cannot contain '%' and '+' characters

error: component contains invalid URI characters
  --> $DIR/route-path-bad-syntax.rs:47:11
   |
47 | #[get("/a?a%20b")] //~ ERROR invalid URI characters
   |           ^^^^^
   |
   = note: components cannot contain '%' and '+' characters

error: component contains invalid URI characters
  --> $DIR/route-path-bad-syntax.rs:51:11
   |
51 | #[get("/a?a+b")] //~ ERROR invalid URI characters
   |           ^^^
   |
   = note: components cannot contain '%' and '+' characters

error: unused dynamic parameter
  --> $DIR/route-path-bad-syntax.rs:57:9
   |
57 | #[get("/<name>")] //~ ERROR unused dynamic parameter
   |         ^^^^^^
   |
note: expected argument named `name` here
  --> $DIR/route-path-bad-syntax.rs:58:7
   |
58 | fn h0(_name: usize) {} //~ NOTE expected argument named `name` here
   |       ^^^^^^^^^^^^

error: unused dynamic parameter
  --> $DIR/route-path-bad-syntax.rs:60:11
   |
60 | #[get("/a?<r>")] //~ ERROR unused dynamic parameter
   |           ^^^
   |
note: expected argument named `r` here
  --> $DIR/route-path-bad-syntax.rs:61:1
   |
61 | fn h1() {} //~ NOTE expected argument named `r` here
   | ^^^^^^^^^^

error: unused dynamic parameter
  --> $DIR/route-path-bad-syntax.rs:63:22
   |
63 | #[post("/a", data = "<test>")] //~ ERROR unused dynamic parameter
   |                      ^^^^^^
   |
note: expected argument named `test` here
  --> $DIR/route-path-bad-syntax.rs:64:1
   |
64 | fn h2() {} //~ NOTE expected argument named `test` here
   | ^^^^^^^^^^

error: unused dynamic parameter
  --> $DIR/route-path-bad-syntax.rs:66:9
   |
66 | #[get("/<_r>")] //~ ERROR unused dynamic parameter
   |         ^^^^
   |
note: expected argument named `_r` here
  --> $DIR/route-path-bad-syntax.rs:67:1
   |
67 | fn h3() {} //~ NOTE expected argument named `_r` here
   | ^^^^^^^^^^

error: unused dynamic parameter
  --> $DIR/route-path-bad-syntax.rs:69:9
   |
69 | #[get("/<_r>/<b>")] //~ ERROR unused dynamic parameter
   |         ^^^^
   |
note: expected argument named `_r` here
  --> $DIR/route-path-bad-syntax.rs:71:1
   |
71 | fn h4() {} //~ NOTE expected argument named `_r` here
   | ^^^^^^^^^^

error: unused dynamic parameter
  --> $DIR/route-path-bad-syntax.rs:69:14
   |
69 | #[get("/<_r>/<b>")] //~ ERROR unused dynamic parameter
   |              ^^^
   |
note: expected argument named `b` here
  --> $DIR/route-path-bad-syntax.rs:71:1
   |
71 | fn h4() {} //~ NOTE expected argument named `_r` here
   | ^^^^^^^^^^

error: `foo_.` is not a valid identifier
  --> $DIR/route-path-bad-syntax.rs:76:9
   |
76 | #[get("/<foo_.>")] //~ ERROR `foo_.` is not a valid identifier
   |         ^^^^^^^
   |
   = help: parameter names must be valid identifiers

error: `foo*` is not a valid identifier
  --> $DIR/route-path-bad-syntax.rs:80:9
   |
80 | #[get("/<foo*>")] //~ ERROR `foo*` is not a valid identifier
   |         ^^^^^^
   |
   = help: parameter names must be valid identifiers

error: `!` is not a valid identifier
  --> $DIR/route-path-bad-syntax.rs:84:9
   |
84 | #[get("/<!>")] //~ ERROR `!` is not a valid identifier
   |         ^^^
   |
   = help: parameter names must be valid identifiers

error: `name>:<id` is not a valid identifier
  --> $DIR/route-path-bad-syntax.rs:88:9
   |
88 | #[get("/<name>:<id>")] //~ ERROR `name>:<id` is not a valid identifier
   |         ^^^^^^^^^^^
   |
   = help: parameter names must be valid identifiers

error: malformed parameter
  --> $DIR/route-path-bad-syntax.rs:94:20
   |
94 | #[get("/", data = "foo")] //~ ERROR malformed parameter
   |                    ^^^
   |
   = help: parameter must be of the form '<param>'

error: malformed parameter
  --> $DIR/route-path-bad-syntax.rs:98:20
   |
98 | #[get("/", data = "<foo..>")] //~ ERROR malformed parameter
   |                    ^^^^^^^
   |
   = help: parameter must be of the form '<param>'

error: parameter is missing a closing bracket
   --> $DIR/route-path-bad-syntax.rs:102:20
    |
102 | #[get("/", data = "<foo")] //~ ERROR missing a closing bracket
    |                    ^^^^
    |
    = help: did you mean '<foo>'?

error: `test ` is not a valid identifier
   --> $DIR/route-path-bad-syntax.rs:106:20
    |
106 | #[get("/", data = "<test >")] //~ ERROR `test ` is not a valid identifier
    |                    ^^^^^^^
    |
    = help: parameter names must be valid identifiers

error: parameters must be named
   --> $DIR/route-path-bad-syntax.rs:112:9
    |
112 | #[get("/<_>")] //~ ERROR must be named
    |         ^^^
    |
    = help: use a name such as `_guard` or `_param`

error: parameter names cannot be empty
   --> $DIR/route-path-bad-syntax.rs:117:9
    |
117 | #[get("/<>")] //~ ERROR cannot be empty
    |         ^^

error: malformed parameter or identifier
   --> $DIR/route-path-bad-syntax.rs:120:9
    |
120 | #[get("/<id><")] //~ ERROR malformed parameter
    |         ^^^^^
    |
    = help: parameters must be of the form '<param>'
    = help: identifiers cannot contain '<' or '>'

error: malformed parameter or identifier
   --> $DIR/route-path-bad-syntax.rs:125:9
    |
125 | #[get("/<<<<id><")] //~ ERROR malformed parameter
    |         ^^^^^^^^
    |
    = help: parameters must be of the form '<param>'
    = help: identifiers cannot contain '<' or '>'

error: malformed parameter or identifier
   --> $DIR/route-path-bad-syntax.rs:130:9
    |
130 | #[get("/<>name><")] //~ ERROR malformed parameter
    |         ^^^^^^^^
    |
    = help: parameters must be of the form '<param>'
    = help: identifiers cannot contain '<' or '>'

error: aborting due to 31 previous errors

